Neo4j.Driver/Neo4j.Driver/Internal/Connector/SocketConnection.cs

```diff
@@ -58,7 +71,11 @@ public void Init()
         {
             try
             {
-                Task.Run(() => _client.Start()).Wait();
+                var connected = Task.Run(() => _client.Start()).Wait(_connectionTimeout);
```

Comment:
We will close the whole connection (where we call `_client.Dispose()`) in `ConnectionPool.Acquire` if failed to create a new connection.